androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/ManageDataLauncherActivity.java

```diff
@@ -294,4 +302,55 @@ public void onCustomTabsServiceConnected(ComponentName componentName,
         @Override
         public void onServiceDisconnected(ComponentName componentName) {}
     }
+
+    /**
+     * Checks whether the provided package name supports site settings. Returns true when
+     * the package name belongs to a twa provider supporting the site settings category or
+     * it is a legacy version of chrome that supports site settings but does not define
+     * the site settings category yet.
+     */
+    public static boolean packageSupportsSiteSettings(String packageName,
+            PackageManager packageManager) {
+        if (packageName == null) return false;
+        if (ChromeLegacyUtils.supportsSiteSettings(packageManager, packageName)) {
+            // Legacy Chrome supports site settings but does not have the site settings category.
+            return true;
+        } else {
+            Intent customTabsIntent = new Intent(CustomTabsService.ACTION_CUSTOM_TABS_CONNECTION);
+            customTabsIntent.addCategory(CATEGORY_LAUNCH_SITE_SETTINGS);
+            customTabsIntent.setPackage(packageName);
+            List<ResolveInfo> services = packageManager.queryIntentServices(customTabsIntent,
+                    PackageManager.GET_RESOLVED_FILTER);
+            return services.size() > 0;
+        }
+    }
+
+    /**
+     * Returns the {@link ShortcutInfo} for a dynamic shortcut into site settings,
+     * provided that {@link ManageDataLauncherActivity} is present in the manifest
+     * and an Intent for managing site settings is available.
+     *
+     * Otherwise returns null if {@link ManageDataLauncherActivity} is not launchable
+     * or if shortcuts are not supported by the Android SDK version.
+     */
+    public static ShortcutInfo getSiteSettingsShortcutOrNull(Context context, PackageManager packageManager) {
+        Intent siteSettingsIntent = new Intent(context, ManageDataLauncherActivity.class);
+        siteSettingsIntent.setAction(ACTION_MANAGE_TRUSTED_WEB_ACTIVITY_DATA);
+        List<ResolveInfo> activities = packageManager.queryIntentActivities(siteSettingsIntent,
+                PackageManager.MATCH_DEFAULT_ONLY);
+        if(activities.size() > 0) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
+                return new ShortcutInfo.Builder(context, SITE_SETTINGS_SHORTCUT_ID)
+                        .setShortLabel("Site Settings")
```

Comment:
We should use string resources here (although we'll need to check whether or not they're included, look at how ChromeUpdatePrompt does it).